### PR TITLE
Minor Improvements to compile_statement, requested during a cc lecture

### PR DESCRIPTION
**Pull request by CAS-ual-TY**
Hi,
during one of your lectures you requested someone (or rather: suggested to do it for training) to combine and simplify 2 different branches: Assignments to dereferenced pointers and to dereferenced expressions.
I have done this quickly and did some basic tests. I can do anymore changes if needed and I can also (try to) combine it further with the non-dereferenced assignments.

### CODE REVIEW

File: selfie.c [hunk 1]
```diff
@@ -5037,8 +5039,25 @@ void compile_statement() {
...
-      // "*" variable "="
+      if (ltype != UINT64STAR_T)
+        type_warning(UINT64STAR_T, ltype);
+
+      if (symbol == SYM_RPARENTHESIS)
+        get_symbol();
+      else
+        syntax_error_symbol(SYM_RPARENTHESIS);
+    } else
+      syntax_error_symbol(SYM_LPARENTHESIS);
```
Comment (ckirsch) [inline selfie.c:5056]: Put a `return` statement in this `else` case and then remove the subsequent check of `ltype`.

File: selfie.c [hunk 2]
```diff
@@ -5014,6 +5014,8 @@ void compile_statement() {
   uint64_t* entry;
   uint64_t offset;
 
+  ltype = 0;
```
Comment (ckirsch) [inline selfie.c:5017]: Remove, see below.

**Review by ckirsch — CHANGES_REQUESTED** [review]
@CAS-ual-TY Nice! Check if my suggestions work, please.

**Review by CAS-ual-TY — COMMENTED** [review]
Done

File: selfie.c [hunk 3]
```diff
@@ -5072,37 +5046,37 @@ void compile_statement() {
       if (ltype != UINT64STAR_T)
         type_warning(UINT64STAR_T, ltype);
 
-      if (symbol == SYM_RPARENTHESIS) {
+      if (symbol == SYM_RPARENTHESIS)
         get_symbol();
+      else
+        syntax_error_symbol(SYM_RPARENTHESIS);
+    } else
+      return;
```
Comment (ckirsch) [inline selfie.c:5054]: Now you are missing reporting the syntax error here…

**Review by ckirsch — CHANGES_REQUESTED** [review]

**Review by CAS-ual-TY — COMMENTED** [review]
Woops, I was a bit too quick there

**Pull request merged**